### PR TITLE
Remove nanoseconds when reading from CSV & writing to parquet

### PR DESCRIPTION
**Pull request by urihoenig**

### CODE REVIEW

File: storey/writers.py [hunk 1]
```diff
@@ -373,6 +373,13 @@ async def _emit(self, batch, batch_key, batch_time):
         if dir_path:
             self._file_system.makedirs(dir_path, exist_ok=True)
         file_path = self._path if self._single_file_mode else f'{dir_path}{uuid.uuid4()}.parquet'
+        # Remove nanosecs from timestamp
+        for name, _ in df.items():
+            if pd.core.dtypes.common.is_datetime64_dtype(df[name]):
+                df[name] = df[name].astype('datetime64[us]')
+        # TODO: Remove or uncomment once keys issue is verifed
```
Comment (gtopper) [inline storey/writers.py]: What issue?

File: storey/writers.py [hunk 2]
```diff
@@ -373,6 +373,13 @@ async def _emit(self, batch, batch_key, batch_time):
         if dir_path:
             self._file_system.makedirs(dir_path, exist_ok=True)
         file_path = self._path if self._single_file_mode else f'{dir_path}{uuid.uuid4()}.parquet'
+        # Remove nanosecs from timestamp
+        for name, _ in df.items():
+            if pd.core.dtypes.common.is_datetime64_dtype(df[name]):
+                df[name] = df[name].astype('datetime64[us]')
```
Comment (gtopper) [inline storey/writers.py]: Shouldn't we be doing this at the source? (DataframeSource)
At least for the time column.

File: tests/test_flow.py [hunk 3]
```diff
@@ -1866,17 +1885,24 @@ def test_csv_reader_parquet_write_microsecs(tmpdir):
...
     assert read_back_df.equals(expected), f"{read_back_df}\n!=\n{expected}"
 
+def test_csv_reader_parquet_write_nanosecs(tmpdir):
+    out_file = f'{tmpdir}/test_csv_reader_parquet_write_nanosecs_{uuid.uuid4().hex}/'
+    columns = ['k', 't']
 
-def test_csv_reader_bad_timestamp(tmpdir):
+    time_format = '%d/%m/%Y %H:%M:%S.%f'
     controller = build_flow([
-        ReadCSV('tests/test-with-timestamp-bad.csv', header=True, key_field='k',
```
Comment (gtopper) [inline tests/test_flow.py:1872]: If you delete the test, I guess you should delete the csv file as well?

File: storey/writers.py (hunk 2 above)
Comment (urihoenig) [reply storey/writers.py]: I placed it first there, but moved it here per OrZ's advice (not to lose data if it's not necessary).

**Review by gtopper — APPROVED** [review]

**Pull request merged**